buildtools/maven/maven.go

```diff
@@ -59,27 +60,75 @@ func (m *Maven) Compile(dir string) error {
 	return err
 }
 
-func Modules(dir string) ([]string, error) {
-	var pom Manifest
-	err := files.ReadXML(&pom, filepath.Join(dir, "pom.xml"))
+// A MvnModule can identify a Maven project with a groupId:artifactId string, a nested
+type MvnModule struct {
+	// Name is taken from the module's POM file.
+	Name string
+
+	// Target is the relative path from the root of the FOSSA project to the module's manifest file or to
+	// the directory containing the module's "pom.xml". The target may name a file other than "pom.xml" and
+	// it may even be the groupId:artifactId string identifying the "Maven project".
+	Target string
+
+	// Path is the relative path from the root of the FOSSA project to the module.
+	Dir string
+
+	GroupId    string
+	ArtifactId string
+}
+
+// Modules returns a list specifying the Maven module at path, which may name a file or directory, and all the
+// Maven modules nested below it. The Target field of each MvnModule is set to the directory or manifest file
+// describing the module. The visited manifest files are listed in the checked map.
+func Modules(path string, checked map[string]bool) ([]MvnModule, error) {
+	cwd, err := os.Getwd()
```

Comment:
I think we want to keep the base directory as an argument to `Modules`, rather than getting cwd

Reply:
I was considering doing it that way too for performance. On the other hand it makes `Modules` more self-contained. If you also think pass it in as an argument, I'd be happy to refactor that